frontend/src/views/DifferentialExpression/components/Main/components/DeResults/components/DifferentialExpressionResults/components/QueryGroupTags/index.tsx

```diff
@@ -28,27 +36,65 @@ const QueryGroupTags = ({
       {nonEmptyQueryGroupKeys.map((key) => {
         const queryGroupKey = key as keyof QueryGroup;
         const selected = queryGroup[queryGroupKey];
+        const selectedId = queryGroupIds[queryGroupKey];
         const suffix = QUERY_GROUP_KEY_TO_TAG_SUFFIX_MAP[queryGroupKey];
-        const label = `${selected.length} ${
-          selected.length > 1 ? suffix.plural : suffix.single
-        }`;
+        const label =
+          selected.length > 1 ? `${selected.length} ${suffix}` : selected[0];
+
+        const getValue = (index: number) => {
+          return key === "cellTypes" ? (
+            <Link
+              label={selected[index]}
+              url={getCellTypeLink({
+                cellTypeId: selectedId[index],
+                tissueId: NO_ORGAN_ID,
+              })}
+            />
+          ) : (
+            selected[index]
+          );
+        };
+        const clickToViewText = "Click to view in CellGuide";
+        const tooltipContent =
+          selected.length === 1 && key === "cellTypes" ? (
+            clickToViewText
+          ) : (
+            <div>
+              <b>{clickToViewText}</b>
+              {selected.map((value, index) => (
+                <div key={`value-${value}-${index}`}>{getValue(index)}</div>
+              ))}
+            </div>
+          );
+
         return (
           <Tooltip
             key={`${key}-tooltip`}
             sdsStyle="light"
             placement="top"
             width="wide"
             leaveDelay={0}
-            title={selected.map((value, index) => (
-              <div key={`value-${value}-${index}`}>{value}</div>
-            ))}
+            disableHoverListener={key !== "cellTypes" && selected.length === 1}
+            title={tooltipContent}
           >
             <span>
               <StyledTag
                 key={key}
                 color="neutral"
                 sdsType="secondary"
+                isSingleCellType={key === "cellTypes" && selected.length === 1}
```

Comment:
extract `const isSingleCellType = key === "cellTypes" && selected.length === 1` and reuse to keep the code DRY